ding/torch_utils/network/res_block.py

```diff
@@ -138,4 +143,6 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         x = self.fc1(x)
         x = self.fc2(x)
         x = self.act(x + identity)
+        if hasattr(self, 'dropout'):
```

Comment:
reduce the usage of `hasattr`, which is not compatible with IDE's code hint. You can define `self.dropout = None` in the constructor function.